client/src/game/ui/notes.ts

```diff
@@ -195,19 +195,13 @@ export const checkNoteImpossibility = (variant: Variant, cardState: CardState, n
   // Validate that the note does not contain an impossibility
   if (note.suitIndex !== null && note.rank === null) {
     // Only the suit was specified
-    // (this logic is copied from the "HanabiCard.checkPipPossibilities()" function)
-    let suitPossible = false;
-    for (const rank of cardState.rankClueMemory.possibilities) {
-      const count = cardState.possibleCards[note.suitIndex][rank];
-      if (count === undefined) {
-        throw new Error(`The card of Suit: ${note.suitIndex} and Rank: ${rank} does not exist in the possibleCards map.`);
-      }
-      if (count > 0) {
-        suitPossible = true;
-        break;
-      }
-    }
-    if (!suitPossible && cardState.location === globals.playerUs) {
+    if (
+      cardState.possibleCardsFromClues.some(
+        ([suitIndex, rank]) => suitIndex === note.suitIndex
+          && cardState.possibleCardsFromObservation[suitIndex][rank] > 0,
```

Comment:
this should probably become a function in rules/card.ts, possible name: "cardRules.canPossiblyBe(cardState, suitIndex, rank)